pkg/eks/api_test.go

```diff
@@ -153,3 +156,97 @@ func mockDescribeImages(p *mockprovider.MockProvider, amiID string, matcher func
 			},
 		}, nil)
 }
+
+var _ = Describe("Setting Availability Zones", func() {
+	var (
+		provider *mockprovider.MockProvider
+		cfg      *api.ClusterConfig
+	)
+
+	BeforeEach(func() {
+		cfg = api.NewClusterConfig()
+		provider = mockprovider.NewMockProvider()
+	})
+
+	When("the AZs were set as CLI params", func() {
+		When("the given params contain enough AZs", func() {
+			It("sets them as the AZs to be used", func() {
+				err := eks.SetAvailabilityZones(cfg, []string{"us-east-2a", "us-east-2b"}, provider.EC2(), "")
+				Expect(err).NotTo(HaveOccurred())
+			})
+		})
+
+		When("the given params contain too few AZs", func() {
+			It("returns an error", func() {
+				err := eks.SetAvailabilityZones(cfg, []string{"us-east-2a"}, provider.EC2(), "")
+				Expect(err).To(MatchError(ContainSubstring("only 1 zone(s) specified [us-east-2a], 2 are required (can be non-unique)")))
```

Comment:
huge nit, here and a few other places isn't this an exact error match? So we could remove the contain substring part
```suggestion
				Expect(err).To(MatchError("only 1 zone(s) specified [us-east-2a], 2 are required (can be non-unique)"))
```

could totally be wrong but first glace looks like all the errors are internal, so its an exact error match

Reply:
testing errors and jake - name a more dynamic duo, I'll wait

Reply:
done!